common/src/Microsoft.Azure.IIoT.Agent.Framework/tests/Worker/WorkerSupervisorTests.cs

```diff
@@ -0,0 +1,203 @@
+
+            // Test
+            const int numberOfWorker = 2;
+            agentConfig.SetMaxWorker(numberOfWorker);
+            await Task.Delay(kDefaultDelay); //Internal timer which creates the Worker run every 10 seconds
+            Assert.Equal(numberOfWorker, sut.NumberOfWorkers);
+
+            // clean up
+            await sut.StopAsync();
+        }
+
+        [Fact]
+        public async Task Test_004_NegativeMaxWorker_Expect_UseDefaultValue() {
+
+            using var container = GetAutofacTestConfiguration();
+            var agentConfig = new TestAgentConfigProvider();
+            var loggerMock = new Mock<ILogger>();
+            loggerMock.Setup(l => l.Error(
+                It.Is<string>(s => s.Contains("MaxWorker")), 
+                It.Is<int>(i => i == kDefaultMaxWorker)))
+                .Verifiable();
+
+            var workerSupervisor = new WorkerSupervisor(container.BeginLifetimeScope(), agentConfig, loggerMock.Object);
+
+            // start host process
+            var sut = workerSupervisor as IWorkerSupervisor;
+            Assert.NotNull(sut);
+            await sut.StartAsync();
+
+            // Test
+            const int numberOfWorker = -4;
+            agentConfig.SetMaxWorker(numberOfWorker);
+
+            await Task.Delay(kDefaultDelay); 
+            Assert.Equal(kDefaultMaxWorker, sut.NumberOfWorkers);
+            loggerMock.Verify();
+
+            // clean up
+            await sut.StopAsync();
+        }
+
+        [Fact]
+        public async Task Test_005_IncreaseAndDecreaseMaxWorker_Expect_Exception() {
+
+            using var container = GetAutofacTestConfiguration();
+            var agentConfig = new TestAgentConfigProvider();
+            var loggerMock = new Mock<ILogger>();
+
+            var workerSupervisor = new WorkerSupervisor(container.BeginLifetimeScope(), agentConfig, loggerMock.Object);
+
+            // start host process
+            var sut = workerSupervisor as IWorkerSupervisor;
+            Assert.NotNull(sut);
+            await sut.StartAsync();
+
+            // Test
+            int numberOfWorker = 8; //Increase
+            agentConfig.SetMaxWorker(numberOfWorker);
+            await Task.Delay(kDefaultDelay);
+            Assert.Equal(numberOfWorker, sut.NumberOfWorkers);
+
+            numberOfWorker = 6; //Decrease
+            agentConfig.SetMaxWorker(numberOfWorker);
+            await Task.Delay(kDefaultDelay);
+            Assert.Equal(numberOfWorker, sut.NumberOfWorkers);
+
+            // clean up
+            await sut.StopAsync();
+        }
+
+        private IContainer GetAutofacTestConfiguration() {
+            var containerBuilder = new ContainerBuilder();
+            containerBuilder.RegisterType<TestWorker>().As<IWorker>();
+            var container = containerBuilder.Build();
+
+            return container;
+        }
+
+        private const int kDefaultDelay = 15000; //Internal timer which creates the Worker run every 10 seconds
```

Comment:
Is it possible to make delay of internal timer configurable and use values smaller than 10 seconds for tests ? Now your tests will wait for 75 seconds only because of the timer.

Reply:
Implemented via optional parameter, were the default value is the original value